### DISCUSSION TITLE
Moving the Home Directory

### DISCUSSION THREAD

**Discussion by spflanze** [openzfs/zfs#13565] [Q&A] [answered]
I successfully installed Ubuntu 22.04 according to the instructions at:
https://openzfs.github.io/openzfs-docs/Getting%20Started/Ubuntu/Ubuntu%2022.04%20Root%20on%20ZFS.html
Which I modified for the needs of my hardware. I desire to have the /home directory on my raided HDD drives, but to reduce the odds of things going wrong I created the home directory in the rpool in accordance with those instructions. The root pool: rpool is in the SSD. Now that I am able to boot Ubuntu I desire to move the /home directory from the SSD to the HDD. Pasted below is what I came up with to do that. I got to the point of the second to last command which is:
`zfs set mountpoint=/home.orig rpool/HOME/ubuntu_oqxf1y`
This command fails because the /home directory is busy. When I executed the command:
`lsof | grep /home`
To find out what is keeping that directory busy my screen filled with a long list of processes. My questions:

1) What would happen if I close all those processes? Would the OS still be functional enough to complete moving the /home directory?

2) What would happen if I used the -f option in the above command to force the mountpoint change?

3) Would it be better to skip the idea of a backup /home.orig directory, and simply execute the forced command:
`zfs -f set mountpoint=/home hpool/HOME/ubuntu_$UUIDh`
?

For those who are interested in how I modified the Ubuntu installation instructions, the latest version of my procedure I followed can be downloaded from:
https://www.mediafire.com/file/2w8mdb96tbzslub/Ubuntu_22.04_Root_on_ZFS_Encryption.odt/file
```
8.5  Move /home to the HDD
# This procedure is based on the instructions here:
#  https://www.howtogeek.com/442101/how-to-move-your-linux-home-directory-to-another-hard-drive/
# They have been modified for zfs RAID.

# 8.5.1  # 2.8.2 Create the New Home’s Pool (hpool):
# ZFS native encryption::
zpool create -f \
 -o ashift=12 \
 -o autotrim=on \
 -O encryption=on -O keylocation=prompt -O keyformat=passphrase \
 -O acltype=posixacl -O xattr=sa -O dnodesize=auto \
 -O compression=lz4 \
 -O normalization=formD \
 -O relatime=on \
 -O canmount=off -O mountpoint=/ -R /mnt \
hpool mirror ${HDD1}-part1  ${HDD2}-part1
touch /etc/zfs/zfs-list.cache/bpool
touch /etc/zfs/zfs-list.cache/rpool
touch /etc/zfs/zfs-list.cache/hpool

 8.5.2  Create the New Home’s Mount Point

# Create the mountpoint for the future home:
zfs create -o canmount=off -o mountpoint=none hpool/HOME

# Mount the HDD at the temporary mount point /mnt
# Create random UUID
UUIDh=$(dd if=/dev/urandom bs=1 count=100 2>/dev/null |
    tr -dc 'a-z0-9' | cut -c-6)

# Make the partition that is to be the new home mountable:
zfs create -o mountpoint=/ \
    -o com.ubuntu.zsys:bootfs=yes \
    -o com.ubuntu.zsys:last-used=$(date +%s) hpool/HOME/ubuntu_$UUIDh

# Make sure there is nothing currently in /mnt
ls /mnt

# 8.5.3  Copy Current Home Folder
# Recursively copy current home folder into the temporary home folder
# while preserving ownership and time stamps:
cp -rp /home/* /mnt

# Verify that all has been copied by examining the contents
ls /mnt

# Unmount the new home partition from /mnt
zfs unmount hpool/HOME/ubuntu_$UUIDh

# Change the /home directory name to have a backup if things go wrong:
zfs set mountpoint=/home.orig rpool/HOME/ubuntu_UUID
# Where UUID was defined earlier

# Mount the new home partition on /home
zfs set mountpoint=/home hpool/HOME/ubuntu_$UUIDh
```

**Comment by rlaager** [comment]
If I were you, I’d login as root from a console. That way, nothing in /home will be in use.

**Comment by spflanze** [comment, marked as answer]
I made some changes to my modifications to rlanger 's procedure. This time I put the home directory on the RAIDED  HDD drives from the beginning, rather than attempting to change it from SDD to the HDD after installation. I am now able to boot Ubuntu. The file which documents my modifications, and is at the download link I provided when I opened this thread, has been updated.

I find that Grub's default generic kernel does not work. I get the error message that it cannot find the root pool when the generic is selected. The "22.04” LTS" kernel versions it offers as boot options do work. I attempted to change the default kernel so I do not have make a selection at boot time, and failed. For this issue I will open a new thread.
